### ISSUE TITLE
Profile `r2c run`

### ISSUE THREAD

**Issue by ulziibay** [returntocorp/cli#74]

**Comment by ievans** [comment]
Comment from user: `time r2c run examples/js` in r2c-permissions repo is 42 seconds for me (and it sends ~4 GiB to docker in the process)

**Comment by ulziibay** [comment]
quoting @deifactor's impressive investigation: we’re doing it because we call copy_local_input, which… copies the input to a temp directory, we could speed it up by just bind-mounting the input repository read-only.
I will look into speeding stuff up

**Comment by deifactor** [comment]
So, I think the right thing to do here is possibly: when running locally, remove all 'fetcher' analyzers from the dependency graph, and when running an analyzer that depends on them, just bind-mount the input directory read-only at the appropriate spot.

This does mean that analyzers can't rely on the input being writable, though. I'm not sure if that's the right tradeoff.

**Comment by ulziibay** [comment]
So, looking at the code, when we run `r2c run` on repo X: we are doing the following and many more serially

Within a public/source:
1) Create /tmp/YY
2) copy X to  /tmp/YY
3) Tar  /tmp/YY/X 
4) CP  the tar to /tmp/local-infra/analysis_output/
Then execute the dependent analyzer by
5) CP  the tar back to /tmp/YY
6) Untar 
7) Run the docker container
8) Interrupt every 1 second to collect stats
This is crazy! I am going to 
1) Disable stat collection 
2) Per @deifactor suggestion, look into better mounting output of fetcher analyzer.

**Comment by deifactor** [comment]
I don't think this is fully fixed; IMO the only way to *actually* fix this would be to move to a world where we just mount the input directory read-only.

**Comment by ulziibay** [comment]
@deifactor: thanks. regarding mounting, is there a recommended python util that mounts a fb directory to a location? `from sh import mount` is that recommended, please advise

Regarding: profiling for stat collection, i did do that minimally in the reverted PR. It slows down marginally: `~30 sec` difference without it for `lodash`. My hunch is that it matters because we are interrupting that docker container and reloading it every second!

**Comment by deifactor** [comment]
I'm saying that instead of copying the Docker files to a new directory and then bind-mounting the new directory when we start up the Docker container, we just bind-mount the original code directory.
